FSharpLu.Json/Compact.fs

```diff
@@ -56,22 +63,28 @@ open ConverterHelpers
 /// The default formatting used by Json.Net to serialize F# discriminated unions
 /// and Option types is too verbose. This module implements a more succinct serialization
 /// for those data types.
-type CompactUnionJsonConverter() =
+type CompactUnionJsonConverter(?tupleAsHeterogeneousArray:bool) =
```

Comment:
I noticed this pattern (define F# optional input, use `defaultArg`, use value) and thought that this could be replaced with:

```fsharp
open System.Runtime.InteropServices

type CompactUnionJsonConverter([<Optional; DefaultValue(true)>] tupleAsHeterogeneousArray) = ...
```

But `DefaultValue` is not actually supported on constructors in F#! That might be a hole in the feature, so I've created a bug here: https://github.com/Microsoft/visualfsharp/issues/5396